FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -20,22 +20,28 @@ class DefaultClientIO : IClientIO
         readonly HttpClient client;
         readonly AuthenticationHeaderValue authHeader;
 
-        internal DefaultClientIO(HttpClient client, AuthenticationHeaderValue authHeader)
+        private LastSeen lastSeen;
+
+        internal DefaultClientIO(HttpClient client, AuthenticationHeaderValue authHeader, LastSeen lastSeen)
         {
             this.client = client;
             this.authHeader = authHeader;
+            this.lastSeen = lastSeen;
         }
 
         public DefaultClientIO(string secret, Uri endpoint, TimeSpan timeout)
-            : this(CreateClient(endpoint, timeout), AuthHeader(secret))
+            : this(CreateClient(endpoint, timeout), AuthHeader(secret), new LastSeen())
         { }
 
         public IClientIO NewSessionClient(string secret) =>
-            new DefaultClientIO(client, AuthHeader(secret));
+            new DefaultClientIO(client, AuthHeader(secret), lastSeen);
 
         public Task<RequestResult> DoRequest(HttpMethodKind method, string path, string data, IReadOnlyDictionary<string, string> query = null) =>
             DoRequestAsync(method, path, data, query);
 
+        public void UpdateLastSeen(long time) =>
```

Comment:
Can this be private or internal?

Reply:
It's an interface method, called from the client - I think it needs to be public?

Reply:
you don't need to expose this method on the interface, you can do all the job on `DefaultClientIO`

Reply:
Aha. So I can. Coming right up.